keyboards/system76/launch_1/launch_1.c

```diff
@@ -0,0 +1,205 @@
+/* 70 */ {15, 0}, {30, 0}, {46, 0}, {61, 0}, {76, 0}, {91, 0}, {106, 0}, {121, 0}, {137, 0}, {152, 0},
+/* 80 */ {167, 0}, {182, 0}, {201, 0}, {224, 0}
+}, {
+    // LED index to flags (set all to LED_FLAG_KEYLIGHT)
+    /*   0  1  2  3  4  5  6  7  8  9 */
+/* 00 */ 4, 4, 4, 4, 4, 4, 4, 4, 4, 4,
+/* 10 */ 4, 4, 4, 4, 4, 4, 4, 4, 4, 4,
+/* 20 */ 4, 4, 4, 4, 4, 4, 4, 4, 4, 4,
+/* 30 */ 4, 4, 4, 4, 4, 4, 4, 4, 4, 4,
+/* 40 */ 4, 4, 4, 4, 4, 4, 4, 4, 4, 4,
+/* 50 */ 4, 4, 4, 4, 4, 4, 4, 4, 4, 4,
+/* 60 */ 4, 4, 4, 4, 4, 4, 4, 4, 4, 4,
+/* 70 */ 4, 4, 4, 4, 4, 4, 4, 4, 4, 4,
+/* 80 */ 4, 4, 4, 4
+} };
+#endif // RGB_MATRIX_ENABLE
+
+bool eeprom_is_valid(void) { 
+    return (
+        eeprom_read_word(((void *)EEPROM_MAGIC_ADDR)) == EEPROM_MAGIC &&
+        eeprom_read_byte(((void *)EEPROM_VERSION_ADDR)) == EEPROM_VERSION
+    );
+}
+// clang-format on
+
+void eeprom_set_valid(bool valid) {
+    eeprom_update_word(((void *)EEPROM_MAGIC_ADDR), valid ? EEPROM_MAGIC : 0xFFFF);
+    eeprom_update_byte(((void *)EEPROM_VERSION_ADDR), valid ? EEPROM_VERSION : 0xFF);
+}
+
+void bootmagic_lite_reset_eeprom(void) {
+    // Set the keyboard-specific EEPROM state as invalid
+    eeprom_set_valid(false);
+    // Set the TMK/QMK EEPROM state as invalid
+    eeconfig_disable();
+}
+
+void system76_ec_rgb_eeprom(bool write);
+void system76_ec_rgb_layer(layer_state_t layer_state);
+void system76_ec_unlock(void);
+bool system76_ec_is_unlocked(void);
+
+rgb_config_t layer_rgb[DYNAMIC_KEYMAP_LAYER_COUNT];
+
+void matrix_init_kb(void) {
+    usb_mux_init();
+
+    bootmagic_lite();
+    if (!eeprom_is_valid()) {
+        dynamic_keymap_reset();
+        dynamic_keymap_macro_reset();
+        system76_ec_rgb_eeprom(true);
+        eeprom_set_valid(true);
+    } else {
+        system76_ec_rgb_eeprom(false);
+    }
+
+    system76_ec_rgb_layer(layer_state);
+}
+
+void matrix_scan_kb(void) {
+    usb_mux_event();
+
+    matrix_scan_user();
+}
+
+#define LEVEL(value) (uint8_t)(((uint16_t)value) * ((uint16_t)RGB_MATRIX_MAXIMUM_BRIGHTNESS) / ((uint16_t)255))
+
+// clang-format off
+static const uint8_t levels[] = {
+    LEVEL(48),
+    LEVEL(72),
+    LEVEL(96),
+    LEVEL(144),
+    LEVEL(192),
+    LEVEL(255)
+};
+// clang-format on
+
+static uint8_t toggle_level = RGB_MATRIX_MAXIMUM_BRIGHTNESS;
+extern bool    input_disabled;
+
+static void set_value_all_layers(uint8_t value) {
+    if (!system76_ec_is_unlocked()) {
+        for (int8_t layer = 0; layer < DYNAMIC_KEYMAP_LAYER_COUNT; layer++) {
+            layer_rgb[layer].hsv.v = value;
+        }
+        system76_ec_rgb_layer(layer_state);
+    }
+}
+
+bool process_record_kb(uint16_t keycode, keyrecord_t *record) {
+    if (input_disabled) { return false; }
+
+    switch (keycode) {
+        case RESET:
+            if (record->event.pressed) {
+                system76_ec_unlock();
+            }
+            return false;
```

Comment:
This code completely overrides the standard handling for the `RESET` keycode in QMK (and also in a way that is not customizable, because this is done before the `process_record_user()` call). It makes the `RESET` keycode useless for someone who wants to use just the upstream QMK without the System76-specific app (`system76_ec_unlock()` just enables the `CMD_RESET` HID command, and then you need to use that app to actually reboot the board into the bootloader).

I wonder whether all those System76 customizations should actually go into a “vendor” keymap instead (some code will probably need to live at the keyboard level, but behind an `#ifdef`).

Reply:
I might need some help to figure out where the `#ifdef`s would need to go if that is the route you want to take.